voltixApp/VoltixApp/View Models/SendCryptoViewModel.swift

```diff
@@ -0,0 +1,345 @@
+//			}
+//			
+//			DispatchQueue.main.async {
+//				self.updateState(tx: tx, eth: eth, web3Service: web3Service)
+//				self.isLoading = false
+//			}
+//		}
+//    }
+    
+    func validateAddress(tx: SendTransaction, address: String) {
+		let chainName = tx.coin.chain.name.lowercased().replacingOccurrences(of: "-", with: "")
+		
+		guard let coinType = CoinType.from(string: chainName) else {
+			print("Coin type not found on Wallet Core")
+			return
+		}
+		
+		isValidAddress = coinType.validate(address: address)
+    }
+    
+    func validateForm(tx: SendTransaction, eth: EthplorerAPIService) -> Bool {
+			// Reset validation state at the beginning
+		errorMessage = ""
+		isValidForm = true
+		
+			// Validate the "To" address
+		if !isValidAddress {
+			errorMessage += "Please enter a valid address. \n"
+			logger.log("Invalid address.")
+			isValidForm = false
+		}
+		
+		let amount = tx.amountDecimal
+		let gasFee = tx.gasDecimal
+		
+		if amount <= 0 {
+			errorMessage += "Amount must be a positive number. Please correct your entry. \n"
+			logger.log("Invalid or non-positive amount.")
+			isValidForm = false
+			return isValidForm
+		}
+		
+		if gasFee <= 0 {
+			errorMessage += "Fee must be a non-negative number. Please correct your entry. \n"
+			logger.log("Invalid or negative fee.")
+			isValidForm = false
+			return isValidForm
+		}
+		
+		let coinName = tx.coin.chain.name.lowercased()
+		let key: String = "\(tx.fromAddress)-\(coinName)"
+		
+		if  tx.coin.chain.chainType == ChainType.UTXO {
+			let walletBalanceInSats = utxo.blockchairData[key]?.address?.balance ?? 0
+			let totalTransactionCostInSats = tx.amountInSats + tx.feeInSats
+			print("Total transaction cost: \(totalTransactionCostInSats)")
+			
+			if totalTransactionCostInSats > walletBalanceInSats {
+				errorMessage += "The combined amount and fee exceed your wallet's balance. Please adjust to proceed. \n"
+				logger.log("Total transaction cost exceeds wallet balance.")
+				isValidForm = false
+			}
+			
+		} else if tx.coin.chain.name.lowercased() == Chain.Ethereum.name.lowercased() {
+			let ethBalanceInWei = Int(eth.addressInfo?.ETH.rawBalance ?? "0") ?? 0 // it is in WEI
+			
+			if tx.coin.ticker.uppercased() == "ETH" {
+				if tx.totalEthTransactionCostWei > ethBalanceInWei {
+					errorMessage += "The combined amount and fee exceed your wallet's balance. Please adjust to proceed. \n"
+					logger.log("Total transaction cost exceeds wallet balance.")
+					isValidForm = false
+				}
+				
+			} else {
+				if let tokenInfo = eth.addressInfo?.tokens?.first(where: { $0.tokenInfo.symbol == tx.coin.ticker.uppercased() }) {
+					print("tx.feeInWei \(tx.feeInWei)")
+					print("ethBalanceInWei \(ethBalanceInWei)")
+					
+					print("has eth to pay the fee?  \(tx.feeInWei > ethBalanceInWei)")
+					
+					if tx.feeInWei > ethBalanceInWei {
+						errorMessage += "You must have ETH in to send any TOKEN, so you can pay the fees. \n"
```

Comment:
These error messages will have to put into Localizable.String , please add a task to your list to do so
No need to fix in this PR

Reply:
Yes.